core/app/models/spree/stock/content_item.rb

```diff
@@ -36,6 +36,17 @@ def amount
         price * quantity
       end
 
+      def hash
+        # XOR operation to get a unique combo hash
+        inventory_unit.hash ^ state.hash
+      end
+
+      def eql?(other)
+        other.class == self.class &&
+          other.inventory_unit == inventory_unit &&
+          (!other.state || other.state == state)
```

Comment:
I've read his explanation, but I still don't understand this.  You can see this comment, which illustrates my point: https://github.com/spree/spree/commit/7837fd1640596055b303f702bdef16a3610149bc#commitcomment-9225757

Everything else seems fine to me.